azurecontainerapps.ts

```diff
@@ -611,7 +611,7 @@ export class azurecontainerapps {
         if (!this.util.isNullOrEmpty(environmentVariables)) {
             // The --replace-env-vars flag is only used for the 'update' command,
             // otherwise --env-vars is used for 'create' and 'up'
-            if (this.noIngressUpdate) {
+            if (this.noIngressUpdate || (!this.noIngressUpdate && !this.adminCredentialsProvided)) {
```

Comment:
Nit : Can we re-name `this.adminCredentialsProvided` to `this.isRegistryCredentialsProvided`?
and `thisIngressUpdate` to `shouldUseUpdate` and assign `(!this.noIngressUpdate && !this.adminCredentialsProvided)` to another local variable?